runtime/parachains/src/inclusion/mod.rs

```diff
@@ -955,10 +930,37 @@ impl<T: Config> CandidateCheckContext<T> {
 	pub(crate) fn verify_backed_candidate(
 		&self,
 		parent_hash: <T as frame_system::Config>::Hash,
+		parent_storage_root: T::Hash,
 		candidate_idx: usize,
 		backed_candidate: &BackedCandidate<<T as frame_system::Config>::Hash>,
 	) -> Result<(), Error<T>> {
 		let para_id = backed_candidate.descriptor().para_id;
+		let now = <frame_system::Pallet<T>>::block_number();
+		let relay_parent_number = now - One::one();
+
+		{
+			// this should never fail because the para is registered
+			let persisted_validation_data = match crate::util::make_persisted_validation_data::<T>(
+				para_id,
+				relay_parent_number,
+				parent_storage_root,
+			) {
+				Some(l) => l,
+				None => {
+					// We don't want to error out here because it will
+					// brick the relay-chain. So we return early without
+					// doing anything.
+					return Ok(())
```

Comment:
This behavior is not the same as before, because there's no return from the outer function. Check out how I did it in my branch with a double-result.

Reply:
(i.e. this behavior is inaccurate)